### PR TITLE
Periph: enable Network Socket to UART passthrough

### PR DESCRIPTION
**Pull request by magicrub**
Enable Network Socket to UART passthrough

This allows a periph to be a UART to IP/Ethernet converter!

Needs more docs..

### CODE REVIEW

File: Tools/AP_Periph/networking.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    AP_SUBGROUPINFO(passthru[8], "PASS9_", 19, Networking_Periph, Passthru),
+#endif
+
+    AP_GROUPEND
+};
+
+
+const AP_Param::GroupInfo Networking_Periph::Passthru::var_info[] = {
+    AP_GROUPINFO_FLAGS("ENABLE", 1,  Networking_Periph::Passthru, enabled, 0, AP_PARAM_FLAG_ENABLE),
+    AP_GROUPINFO("EP1", 2,  Networking_Periph::Passthru, ep1, -1),
```
Comment (tridge) [inline Tools/AP_Periph/networking.cpp:150]: needs param doc markup

**Review by tridge — CHANGES_REQUESTED** [review]
looks good! 
needs some param docs, plus I suspect you'll need the passthru object param table to be in a separate file to pass the param generation CI test?

File: Tools/AP_Periph/networking.cpp (hunk 1 above)
Comment (magicrub) [reply Tools/AP_Periph/networking.cpp:163]: param docs added! I'll see what CI says about if I need them in a separate file or not

**Comment by magicrub** [comment]
I tested, and fixed, the docs using Peter's PR https://github.com/ArduPilot/ardupilot/pull/25613 which really helped.

**Review by Ryanf55 — CHANGES_REQUESTED** [review]

File: Tools/AP_Periph/networking.cpp [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+extern const AP_HAL::HAL &hal;
+
+const AP_Param::GroupInfo Networking_Periph::var_info[] {
+    // @Group:
+    // @Path: ../AP_Networking/AP_Networking.cpp
+    AP_SUBGROUPINFO(networking_lib, "", 1, Networking_Periph, AP_Networking),
+
+    /*
+      the NET_Pn_ parameters need to be bere as otherwise we
```
[the remark this answers is not in the document]
Comment (magicrub) [reply Tools/AP_Periph/networking.cpp]: thanks. Fixed!

File: Tools/AP_Periph/networking_passthru.cpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    }
+}
+
+void Networking_Periph::Passthru::update()
+{
+    if (enabled == 0 || port1 == nullptr || port2 == nullptr) {
+        return;
+    }
+
+    // Fastest possible connection is 3Mbps serial port, which is roughly 300KB/s payload and we service this at <= 1kHz
```
Comment (shubham-shahh) [inline Tools/AP_Periph/networking_passthru.cpp:107]: Hi, How do we come to this number of 3Mbps?

File: Tools/AP_Periph/networking_passthru.cpp [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+void Networking_Periph::Passthru::init()
+{
+    if (enabled == 0) {
+        // Feature is disabled
+        return;
+    }
+
+    if (port1 != nullptr || port2 != nullptr) {
+        // Endpoints have already been initilized
```
Comment (shubham-shahh) [inline Tools/AP_Periph/networking_passthru.cpp:80]: small error, initialized

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@magicrub looks good to me, just need to get it all OK in CI and param gen

File: Tools/AP_Periph/networking_passthru.cpp (hunk 3 above)
Comment (magicrub) [reply Tools/AP_Periph/networking_passthru.cpp:107]: UARTs can't really go higher than that reliably. Is there a different number you're thinking? Bigger/smaller?

Comment (shubham-shahh) [reply Tools/AP_Periph/networking_passthru.cpp:107]: > UARTs can't really go higher than that reliably. Is there a different number you're thinking? Bigger/smaller?

No, I was just wondering if there is any theoretical way to derive it.

File: Tools/AP_Periph/networking_passthru.cpp (hunk 4 above)
Comment (magicrub) [reply Tools/AP_Periph/networking_passthru.cpp]: Thanks. Changed to "// The ports have already been initialized, nothing to do."

**Pull request merged**